taskvine/src/worker/vine_cache.c

```diff
@@ -69,9 +70,62 @@ struct vine_cache * vine_cache_create( const char *cache_dir )
 }
 
 /*
-Delete the cache manager structure, though not the underlying files.
+Load exsiting cache directory into cache structure.
 */
+void vine_cache_load(struct vine_cache *c)
+{
+	DIR *dir = opendir(c->cache_dir);
+	if(dir){
+		debug(D_VINE, "loading cache at: %s", c->cache_dir);
+		struct dirent *d;
+		while((d=readdir(dir))){
+			debug(D_VINE, "found %s in cache at: %s",d->d_name, c->cache_dir);
+			struct stat info;
+			char *cache_path = vine_cache_full_path(c,d->d_name);
+			if(stat(cache_path, &info)==0){
+				if(!strncmp(d->d_name, "file", strlen("file"))){
```

Comment:
If the stored object is a directory and not just a plain file, then we need to measure the size recursively, and not just the simple size from info.st_size, right?  (I'm sure we have a function to do that already.)